packages/block-library/src/post-navigation-link/edit.js

```diff
@@ -77,18 +77,12 @@ export default function PostNavigationLinkEdit( {
 			label: __( 'Unfiltered' ),
 			value: '',
 		};
-		const taxonomyOptions = ( taxonomies ?? [] )
-			.filter(
-				( tax ) =>
-					tax.slug !== 'nav_menu' &&
-					tax.slug !== 'wp_pattern_category'
-			)
-			.map( ( item ) => {
-				return {
-					value: item.slug,
-					label: item.name,
-				};
-			} );
+		const taxonomyOptions = ( taxonomies ?? [] ).map( ( item ) => {
+			return {
+				value: item.slug,
+				label: item.name,
+			};
+		} );
```

Comment:
This change is what I suggested in [this comment](https://github.com/WordPress/gutenberg/pull/48912#discussion_r1450604136), but I was a bit misled.

If we make this change, "Navigation Menus" and "Pattern Categories" will appear again in **the Site Editor**. This block is primarily inserted in post templates without context. In other words, `postType` is `undefined` here:

https://github.com/WordPress/gutenberg/blob/3a73b89de28ffcfacd15e202e5fdfb95dc652467/packages/block-library/src/post-navigation-link/edit.js#L28

The fact that `postType` is `undefined` is probably the reason why unintended taxonomies are displayed.

If we don't want to hardcode a specific taxonomy and want to get only the appropriate taxonomies even in the Site Editor, we'll need to filter in a different way.

In my opinion, we should only need to check if `publicly_queryable` is `true`. This approach can also be seen [here](https://github.com/WordPress/gutenberg/blob/3a73b89de28ffcfacd15e202e5fdfb95dc652467/packages/edit-site/src/components/add-new-template/utils.js#L89-L99).

That is, change it to:

```diff
diff --git a/packages/block-library/src/post-navigation-link/edit.js b/packages/block-library/src/post-navigation-link/edit.js
index 464531f9d4..be77a848cc 100644
                                type: postType,
                                per_page: -1,
-                               context: 'view',
                        } );
                        return filteredTaxonomies;
                },
                [ postType ]
        );
+
        const getTaxonomyOptions = () => {
                const selectOption = {
                        label: __( 'Unfiltered' ),
                        value: '',
                };
                const taxonomyOptions = ( taxonomies ?? [] )
-                       .filter(
-                               ( tax ) =>
-                                       tax.slug !== 'nav_menu' &&
-                                       tax.slug !== 'wp_pattern_category'
-                       )
+                       .filter( ( { visibility } ) => !! visibility?.publicly_queryable )
                        .map( ( item ) => {
                                return {
                                        value: item.slug,
```

This approach should still give us the expected behavior:

- In the Post Editor, we will only get taxonomies available for that post type.
- In te Site editor we will get taxonomies available for all posts